From 1db81b4bd377d247dceae65c054f44936c45498a Mon Sep 17 00:00:00 2001
From: Uri Mashiach <uri.mashiach@compulab.co.il>
Date: Tue, 19 Sep 2017 13:39:04 +0300
Subject: [PATCH 22/57] arm: imx7d: cl-som-imx7: add support for eeprom layout
 APIs

Add and initialize layout structures for the SOM end baseboard EEPROMs.

Signed-off-by: Uri Mashiach <uri.mashiach@compulab.co.il>
---
 board/compulab/cl-som-imx7/cl-som-imx7.c | 21 +++++++++++++++++++++
 include/configs/cl-som-imx7.h            |  2 ++
 2 files changed, 23 insertions(+)

diff --git a/board/compulab/cl-som-imx7/cl-som-imx7.c b/board/compulab/cl-som-imx7/cl-som-imx7.c
index 7ca4c4d5a3..ec515b07ec 100644
--- a/board/compulab/cl-som-imx7/cl-som-imx7.c
+++ b/board/compulab/cl-som-imx7/cl-som-imx7.c
@@ -28,6 +28,11 @@ DECLARE_GLOBAL_DATA_PTR;
 
 static int nand_enabled = 0;
 
+static uchar cl_som_am57x_eeprom_buf[CONFIG_SYS_EEPROM_SIZE];
+static uchar sb_som_am57x_eeprom_buf[CONFIG_SYS_EEPROM_SIZE];
+static struct eeprom_layout cl_som_am57x_layout;
+static struct eeprom_layout sb_som_am57x_layout;
+
 #ifdef CONFIG_SYS_I2C_MXC
 
 /* Baseboard I2C bus is initialized flag */
@@ -307,12 +312,28 @@ int board_early_init_f(void)
 
 int board_init(void)
 {
+	int ret;
+
 	/* address of boot parameters */
 	gd->bd->bi_boot_params = PHYS_SDRAM + 0x100;
 	cl_som_imx7_setup_i2c();
 	cl_som_imx7_setup_gpmi_nand();
 	cl_som_imx7_setup_fec();
 	cl_som_imx7_spi_init();
+	ret = cl_eeprom_layout_setup(&cl_som_am57x_layout,
+				     cl_som_am57x_eeprom_buf,
+				     LAYOUT_VERSION_AUTODETECT,
+				     CONFIG_SYS_I2C_EEPROM_BUS,
+				     CONFIG_SYS_I2C_EEPROM_ADDR);
+	if (cl_som_imx7_base_i2c_init)
+		ret |= cl_eeprom_layout_setup(&sb_som_am57x_layout,
+					      sb_som_am57x_eeprom_buf,
+					      LAYOUT_VERSION_AUTODETECT,
+					      CL_SOM_IMX7_I2C_BUS_EXT,
+					      CL_SOM_IMX7_I2C_EEPROM_EXT);
+
+       if (ret)
+               printf("EEPROM layout initialization failure\n");
 
 	return 0;
 }
diff --git a/include/configs/cl-som-imx7.h b/include/configs/cl-som-imx7.h
index 2ede00c19e..650c79a0d6 100644
--- a/include/configs/cl-som-imx7.h
+++ b/include/configs/cl-som-imx7.h
@@ -53,8 +53,10 @@
 #define CONFIG_SYS_I2C_MXC_I2C4		/* Enable I2C bus 4 */
 #define CONFIG_SYS_I2C_SPEED		100000
 #define SYS_I2C_BUS_SOM			0
+#define CL_SOM_IMX7_I2C_BUS_EXT		1
 
 #define CONFIG_SYS_I2C_EEPROM_ADDR	0x50
+#define CL_SOM_IMX7_I2C_EEPROM_EXT	0x54
 #define CONFIG_SYS_I2C_EEPROM_ADDR_LEN	1
 #define CONFIG_SYS_I2C_EEPROM_BUS	SYS_I2C_BUS_SOM
 
-- 
2.17.1

